Controller runtime v0.16.4 Go 1.20

gibizer opened the pull request.

stuggi commented on Makefile:
```diff
@@ -325,3 +325,10 @@ run-with-webhook: export PLACEMENT_API_IMAGE_URL_DEFAULT=quay.io/tripleozedcento
 	go run ./main.go
+
+.PHONY: tidy
+tidy: fmt
```
I called it gotidy in lib-common [1] and also added that as allowed command in hosted runner [2]. might want to have the same here, or rename it in lib-common so that it is in sync

[1] https://github.com/openstack-k8s-operators/lib-common/blob/master/Makefile#L130
[2] https://github.com/openstack-k8s-operators/openstack-k8s-operators-ci/blob/master/renovate.sh#L12

gibizer replied:
The majority of the operators has `make tidy` now https://github.com/search?q=org%3Aopenstack-k8s-operators+%22.PHONY+tidy%22&type=code vs `make gotidy` https://github.com/search?q=org%3Aopenstack-k8s-operators+%22.PHONY+gotidy%22&type=code So if we want to sync then I would rather do the rename in  lib-common

stuggi replied:
ack, haven't seen those . I'll adjust lib-common then

SeanMooney replied:
when we have been addign targets that wrap go commands we have generally been avoiding prefixing with go.

the SDK also avoids that so as a general pattern if we need other go X targets i think we should just call them X rather then goX as a convention.

mrKisaoLamb reviewed (commented):
Shoudn't we also change pre-commit tidy to use make tidy? https://github.com/openstack-k8s-operators/placement-operator/blob/master/.pre-commit-config.yaml#L23

gibizer commented:
> Shoudn't we also change pre-commit tidy to use make tidy? https://github.com/openstack-k8s-operators/placement-operator/blob/master/.pre-commit-config.yaml#L23

I hope there will be a separate PR from @amolkahat similar to what they do for nova-operator to switch to make targets in pre-commit. I just needed the make tidy target from that effort for renovate.

Commit:
[webhook] Update webhook functions for CR update

This change updates the webhook functions to comply with the new
standard for controller-runtime.

Signed-off-by: Brendan Shephard <bshephar@redhat.com>

Commit:
[manager] Update manager initialisation for CR update

This change updates the manager intialisation to comply with the new controller-runtime schema.

Signed-off-by: Brendan Shephard <bshephar@redhat.com>

Commit:
[watches] Fix watches declarations for CR update

This change fixes the Watches declarations to comply with the new
controller-runtime standards

Signed-off-by: Brendan Shephard <bshephar@redhat.com>

gibizer committed:
Update controller-runtime version

gibizer committed:
Update to Go 1.20

gibizer committed:
Bump envtest to 1.28 to match k8s.io

gibizer commented:
We need to force merge all the bump to get back to green CI state.
